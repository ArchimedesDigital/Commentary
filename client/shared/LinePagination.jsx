import RaisedButton from 'material-ui/RaisedButton';

LinePagination = React.createClass({
	propTypes: {
		linePagination: React.PropTypes.array.isRequired,
		linePaginationClicked: React.PropTypes.func.isRequired,
	},

	render() {
		const self = this;

		return (
			<div className="line-pagination">
				{this.props.linePagination.map(function(line, i){
					return (
						<RaisedButton
								key={i}
								label={line}
								className="line-page"
								onClick={self.props.linePaginationClicked.bind(null, line)}
						/>
						);
				})}
			</div>

		);

	}
});
