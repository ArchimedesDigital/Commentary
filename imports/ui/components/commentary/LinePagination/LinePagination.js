import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';

const LinePagination = ({ linePagination, linePaginationClicked }) => (
	<div className="line-pagination">
		{linePagination.map((line, i) => (
			<RaisedButton
				key={line}
				label={line}
				className="line-page"
				onClick={linePaginationClicked.bind(null, line)}
			/>
		))}
	</div>
);
LinePagination.propTypes = {
	linePagination: React.PropTypes.arrayOf(React.PropTypes.number).isRequired,
	linePaginationClicked: React.PropTypes.func.isRequired,
};

export default LinePagination;
