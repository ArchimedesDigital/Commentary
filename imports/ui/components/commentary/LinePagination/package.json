{
  "name": "LinePagination",
  "version": "0.0.1",
  "private": true,
  "main": "./LinePagination.js"
}
