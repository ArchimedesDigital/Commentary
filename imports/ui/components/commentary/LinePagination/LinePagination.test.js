import React from 'react';
import renderer from 'react-test-renderer';

// component:
import LinePagination from './LinePagination';

describe('LinePagination', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<LinePagination linePagination={[]} linePaginationClicked={() => {}} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
