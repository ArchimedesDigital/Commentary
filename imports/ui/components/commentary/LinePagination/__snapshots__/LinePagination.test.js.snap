// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinePagination renders correctly 1`] = `
<div
  className="line-pagination"
/>
`;
